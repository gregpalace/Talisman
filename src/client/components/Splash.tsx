import React, { useEffect } from 'react';

import LoadIcon from '../components/LoadIcon';

import '../../stylesheets/splash.scss';

interface SplashProps {
  setFirstVisit: React.Dispatch<React.SetStateAction<boolean>>;
}

const Splash = ({ setFirstVisit }: SplashProps) => {
  // Display splash for 3 seconds
  useEffect(() => {
    setTimeout(() => setFirstVisit(false), 3000);
  }, []);

  return (
    <div id='splash'>
      <img id='splashLogo' src={'./assets/logo/logo.png'} alt='Talisman' />
      <h1>Talisman</h1>
      <LoadIcon />
    </div>
  );
};

export default Splash;
