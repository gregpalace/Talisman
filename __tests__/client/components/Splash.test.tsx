import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Splash from '../../../src/client/components/Splash';

describe('React unit tests', () => {
  let wrapper: any;

  const props = {
    setFirstVisit: jest.fn(),
  };

  beforeEach(() => {
    wrapper = shallow(<Splash {...props} />);
  });

  describe('<Splash/>', () => {
    it('should render correctly', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
