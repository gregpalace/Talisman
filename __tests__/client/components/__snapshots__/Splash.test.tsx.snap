// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React unit tests <Splash/> should render correctly 1`] = `
<div
  id="splash"
>
  <img
    alt="Talisman"
    id="splashLogo"
    src="./assets/logo/logo.png"
  />
  <h1>
    Talisman
  </h1>
  <LoadIcon />
</div>
`;
